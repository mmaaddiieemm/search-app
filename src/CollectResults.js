import React from 'react';
import Search from './Search';
import Users from './Users.js';
import Tickets from './Tickets.js';
import Organizations from './Organizations.js';

const UsersDb = require('./data/users.json');
const TicketsDb = require('./data/tickets.json');
const OrgDb = require('./data/organizations.json');
 
function CollectResults({ searchKey, value, dataTypes }) {
  // search for all matching key:value pairs in the data and add them 
  // to the results. 
  var users = (dataTypes.includes('u')) ? Search(searchKey, value, UsersDb) : [];
  var tickets = (dataTypes.includes('t')) ? Search(searchKey, value, TicketsDb) : [];
  var orgs = (dataTypes.includes('o')) ? Search(searchKey, value, OrgDb) : [];

  // this block attempts to identify the related items and adds them to the list. 
  // this definitely needs to be updated to utilize some sort of set so that 
  // we don't get duplication of entries. 
  // if ( typeof searchKey !== 'undefined' && typeof value !== 'undefined' && !searchKey.includes('id')) {
  //   // loop through secondary results and add them to the list? 
  //   let firstResults = users.concat(tickets).concat(orgs); 
  //   for (var result in firstResults) {
  //     users = users.concat(Search('id', firstResults[result]._id, UsersDb)); 
  //     tickets = tickets.concat(Search('id', firstResults[result]._id, TicketsDb)); 
  //     orgs = orgs.concat(Search('id', firstResults[result]._id, OrgDb)); 
  //   }    
  // }
  
  if (users.length === 0 && tickets.length === 0 && orgs.length === 0 ) {
    return <p className="App-results">No entry found that matches criteria</p>; 
  } else {
    return (
      <div>
        <Users users={users} />
        <Tickets tickets={tickets} />
        <Organizations orgs={orgs} />
      </div>
    ); 
  }
}

export default CollectResults;
