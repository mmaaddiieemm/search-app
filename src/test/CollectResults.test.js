import React from "react"
import renderer from 'react-test-renderer';
import {cleanup, fireEvent, render} from '@testing-library/react';
import CollectResults from "../CollectResults.js"; 

afterEach(cleanup);

test('When nonsense data is input, return an error message', () => {
    const component = renderer.create(
        <CollectResults searchKey='' value='' dataTypes='' />
    ); 
    let tree = component.toJSON(); 
    expect(tree).toMatchSnapshot(); 
}); 

test('when a specific data type is requested, return only data from the specified datatype', () => {
    const component = renderer.create(
        <CollectResults searchKey="id" value='4' dataTypes='u' />
    ); 
    let tree = component.toJSON(); 
    expect(tree).toMatchSnapshot(); 
}); 

test('when no datatype is specified, return results from all data types', () => {

}); 

test('when valid input is entered and there is matching data, render the relevant data', () => {
    
});

test('when valid input is entered and there is no matching data, render a message indicating no relevant data found', () => {
    
}); 
