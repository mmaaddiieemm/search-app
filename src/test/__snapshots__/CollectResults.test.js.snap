// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When nonsense data is input, return an error message 1`] = `
<div>
  <p
    className="App-results"
  >
    No entry found that matches criteria
  </p>
  <div
    className="Results"
  >
    <div
      className="list-group"
      onKeyDown={[Function]}
    >
      <label
        className="Type"
      >
        
      </label>
    </div>
  </div>
  <div
    className="Results"
  >
    <div
      className="list-group"
      onKeyDown={[Function]}
    >
      <label
        className="Type"
      >
        
      </label>
    </div>
  </div>
  <div
    className="Results"
  >
    <div
      className="list-group"
      onKeyDown={[Function]}
    >
      <label
        className="Type"
      >
        
      </label>
    </div>
  </div>
</div>
`;

exports[`when a specific data type is requested, return only data from the specified datatype 1`] = `
<div>
  <div
    className="Results"
  >
    <div
      className="list-group"
      onKeyDown={[Function]}
    >
      <label
        className="Type"
      >
        Users
      </label>
      <div
        className="list-group-item"
        data-rb-event-key={null}
        disabled={false}
        onClick={[Function]}
      >
        <label>
          <b>
             
            name
            :
             
          </b>
           
          Rose Newton
           
        </label>
        <label>
          <b>
             
            alias
            :
             
          </b>
           
          Mr Cardenas
           
        </label>
        <label>
          <b>
             
            email
            :
             
          </b>
           
          cardenasnewton@flotonic.com
           
        </label>
        <label>
          <b>
             
            url
            :
             
          </b>
           
          http://initech.zendesk.com/api/v2/users/4.json
           
        </label>
        <label>
          <b>
             
            phone
            :
             
          </b>
           
          8685-482-450
           
        </label>
        <label>
          <b>
             
            tags
            :
             
          </b>
           
          [Gallina,Glenshaw,Rowe,Babb]
           
        </label>
        <label>
          <b>
             
            role
            :
             
          </b>
           
          end-user
           
        </label>
        <label>
          <b>
             
            signature
            :
             
          </b>
           
          Don't Worry Be Happy!
           
        </label>
        <label>
          <b>
             
            organization_id
            :
             
          </b>
           
          122
           
        </label>
        <label>
          <b>
             
            active
            :
             
          </b>
           
          true
           
        </label>
        <label>
          <b>
             
            verified
            :
             
          </b>
           
          true
           
        </label>
        <label>
          <b>
             
            suspended
            :
             
          </b>
           
          true
           
        </label>
        <label>
          <b>
             
            shared
            :
             
          </b>
           
          true
           
        </label>
        <label>
          <b>
             
            locale
            :
             
          </b>
           
          de-CH
           
        </label>
        <label>
          <b>
             
            timezone
            :
             
          </b>
           
          Netherlands
           
        </label>
        <label>
          <b>
             
            last_login_at
            :
             
          </b>
           
          2012-09-25T01:32:46 -10:00
           
        </label>
        <label>
          <b>
             
            _id
            :
             
          </b>
           
          4
           
        </label>
        <label>
          <b>
             
            external_id
            :
             
          </b>
           
          37c9aef5-cf01-4b07-af24-c6c49ac1d1c7
           
        </label>
        <label>
          <b>
             
            created_at
            :
             
          </b>
           
          2016-02-09T07:52:10 -11:00
           
        </label>
      </div>
    </div>
  </div>
  <div
    className="Results"
  >
    <div
      className="list-group"
      onKeyDown={[Function]}
    >
      <label
        className="Type"
      >
        
      </label>
    </div>
  </div>
  <div
    className="Results"
  >
    <div
      className="list-group"
      onKeyDown={[Function]}
    >
      <label
        className="Type"
      >
        
      </label>
    </div>
  </div>
</div>
`;
